controllers/azurecluster_reconciler.go

```diff
@@ -44,8 +44,11 @@ type azureClusterService struct {
 	scope *scope.ClusterScope
 	// services is the list of services that are reconciled by this controller.
 	// The order of the services is important as it determines the order in which the services are reconciled.
-	services []azure.ServiceReconciler
-	skuCache *resourceskus.Cache
+	services  []azure.ServiceReconciler
+	skuCache  *resourceskus.Cache
+	Reconcile func(context.Context) error
+	Pause     func(context.Context) error
+	Delete    func(context.Context) error
```

Comment:
Are these changes in the `azurecluster_reconciler` needed for testing? I'm assuming this may have been leftover from a rebase.

Reply:
No these are needed for testing and it now follows the same approach as `azuremachine_reconciler` does

Reply:
to test `getFakeAzureClusterServiceWithGeneralError` [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/8d81f95b1a80fad346dbb4c9424f0ebe9ea845bf/controllers/azurecluster_controller_test.go#L370) we need the approach